backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/document/model/dsl/impl/ElasticsearchIndexSchemaFieldContextImpl.java

```diff
@@ -109,7 +112,7 @@ public EventContext getEventContext() {
 
 	private <T extends ElasticsearchIndexSchemaNodeContributor<PropertyMapping>> T setDelegate(T context) {
 		if ( delegate != null ) {
-			throw new SearchException( "You cannot set the type of a field more than once" );
+			throw log.tryToSetFieldTypeMoreThanOnce( getEventContext() );
```

Comment:
Good idea, some context here and in the exception above could help.

Reply:
thanks!